### DISCUSSION TITLE
Pretty Confused about the cleaning strategy on laravel backup ?

### DISCUSSION THREAD

**Discussion by subhadipghorui** [spatie/laravel-backup#1208] [Q&A]
I am using it on my laravel app. I can't understand how the default cleaning strategy works? I read the documentation but no help.
Can anyone give an example ??

Confused - keep_all_backups_for_days with keep_daily_backups_for_days and keep_weekly_backups_for_weeks

What are daily vs weekly backups?
`'cleanup' => [
        /*
         * The strategy that will be used to cleanup old backups. The default strategy
         * will keep all backups for a certain amount of days. After that period only
         * a daily backup will be kept. After that period only weekly backups will
         * be kept and so on.
         *
         * No matter how you configure it the default strategy will never
         * delete the newest backup.
         */
        'strategy' => \Spatie\Backup\Tasks\Cleanup\Strategies\DefaultStrategy::class,

        'default_strategy' => [

            /*
             * The number of days for which backups must be kept.
             */
            'keep_all_backups_for_days' => 7,

            /*
             * The number of days for which daily backups must be kept.
             */
            'keep_daily_backups_for_days' => 16,

            /*
             * The number of weeks for which one weekly backup must be kept.
             */
            'keep_weekly_backups_for_weeks' => 8,

            /*
             * The number of months for which one monthly backup must be kept.
             */
            'keep_monthly_backups_for_months' => 4,

            /*
             * The number of years for which one yearly backup must be kept.
             */
            'keep_yearly_backups_for_years' => 2,

            /*
             * After cleaning up the backups remove the oldest backup until
             * this amount of megabytes has been reached.
             */
            'delete_oldest_backups_when_using_more_megabytes_than' => 400,
        ],
    ],`

**Comment by freekmurze** [comment]
In the presentation I give some more info: https://youtu.be/fORNQ06K8LY?t=1115

**Comment by subhadipghorui** [comment]
Thanks for the reply.☺

**Comment by reppair** [comment]
Good one. Got me confused as well. But this presentation screen helps, the "grandfather-father-son" backup strategy...

<img width="1477" alt="Screenshot 2024-10-15 at 21 54 08" src="https://github.com/user-attachments/assets/66541a18-b93b-4ed8-b2e2-6291cc6edb3d">
